backend/shared/src/main/scala/zio/logging/LogFormat.scala

```diff
@@ -0,0 +1,214 @@
+    override private[logging] def unsafeFormat(builder: String => Unit) = new ZLogger[Unit] {
+      override def apply(
+        trace: ZTraceElement,
+        fiberId: FiberId,
+        logLevel: LogLevel,
+        message: () => String,
+        context: Map[ZFiberRef.Runtime[_], AnyRef],
+        spans: List[LogSpan]
+      ): Unit = {
+        format(builder, trace, fiberId, logLevel, message, context, spans)
+        ()
+      }
+    }
+
+    override def toLogger: ZLogger[String] = new ZLogger[String] {
+      override def apply(
+        trace: ZTraceElement,
+        fiberId: FiberId,
+        logLevel: LogLevel,
+        message: () => String,
+        context: Map[ZFiberRef.Runtime[_], AnyRef],
+        spans: List[LogSpan]
+      ): String = {
+        val builder = new StringBuilder()
+        //FIXME this should not be required when core will handle exception properly.
+        try unsafeFormat(builder.append(_: String))(trace, fiberId, logLevel, message, context, spans)
+        catch {
+          case ex: Throwable => ex.printStackTrace()
+        }
+
+        builder.toString()
+      }
+    }
+  }
+
+  val bracketStart: LogFormat[String] = string("[")
+
+  val bracketEnd: LogFormat[String] = string("]")
+
+  val quote: LogFormat[String] = string("\"")
+
+  def quoted(inner: LogFormat[String]): LogFormat[String] = quote + inner + quote
+
+  val newLine: LogFormat[String] = string(NL)
+
+  def bracketed(inner: LogFormat[String]): LogFormat[String] =
+    bracketStart + inner + bracketEnd
+
+  val level: LogFormat[String] =
+    LogFormat { (builder, _, _, level, _, _, _) =>
+      builder(level.label)
+    }
+
+  val fiberNumber: LogFormat[String] =
+    LogFormat { (builder, _, fiberId, _, _, _, _) =>
+      builder("#")
+      builder(fiberId.seqNumber.toString)
+    }
+
+  def annotation(name: String): LogFormat[String] =
+    LogFormat { (builder, _, _, _, _, fiberRefs, _) =>
+      fiberRefs
+        .get(logAnnotation)
+        .foreach(value => value.asInstanceOf[Map[String, String]].get(name).foreach(builder(_)))
+    }
+
+  def label(label: String): LogFormat[String] =
+    string(label) + string("=")
+
+  def label(label: String, value: LogFormat[String]): LogFormat[String] =
+    string(label) + string("=") + value
+
+  def string(label: String): LogFormat[String] =
+    LogFormat { (builder, _, _, _, _, _, _) =>
+      builder(label)
+    }
+
+  private def defaultHighlighter(level: LogLevel) = level match {
+    case LogLevel.Error   => LogColor.RED
+    case LogLevel.Warning => LogColor.YELLOW
+    case LogLevel.Info    => LogColor.CYAN
+    case LogLevel.Debug   => LogColor.GREEN
+    case _                => LogColor.WHITE
+  }
+
+  val timestamp: LogFormat[String] = timestamp(DateTimeFormatter.ISO_OFFSET_DATE_TIME)
+
+  def timestamp(formatter: DateTimeFormatter): LogFormat[String] =
+    string {
+      val now = ZonedDateTime.now()
+      formatter.format(now)
+    }
+
+  val line: LogFormat[String] =
```

Comment:
I would order all methods alphabetically, both in the companion object, and inside the `LogFormat` trait.

Reply:
`default` and `colored` need to be at the end because of initialization unless we are ok with `lazy val`